client/core/core.go

```diff
@@ -2497,3 +2497,25 @@ func validateOrderResponse(dc *dexConnection, result *msgjson.OrderResult, ord o
 	}
 	return nil
 }
+
+// Logout logs the user out
+func (c *Core) Logout() error {
+	c.connMtx.RLock()
+	defer c.connMtx.RUnlock()
+	for _, asset := range c.user.Assets {
```

Comment:
The map key for `c.user.Assets` is the asset ID itself, so just range like `for assetID := range c.user.Assets {` and use `assetID` instead of `asset.ID` below.